server/src/test/java/io/crate/integrationtests/PartitionedTableIntegrationTest.java

```diff
@@ -2364,4 +2364,38 @@ public void test_select_partitioned_by_column_with_query_then_fetch_plan() throw
             "1| Trillian\n"
         ));
     }
+
+    @Test
+    @UseRandomizedSchema(random = false)
+    public void test_alter_table_drop_constraint_on_partitioned_table() throws Exception {
+        execute("create table t (id int primary key constraint check_id_gt_zero check(id > 0), " +
+            "x int constraint check_x_gt_zero check(x > 0)) " +
+            "partitioned by (id) " +
+            "clustered into 1 shards " +
+            "with (number_of_replicas=0)");
+        refresh();
+
+        String selectCheckConstraintsStmt =
+            "select table_schema, table_name, constraint_type, constraint_name " +
+                "from information_schema.table_constraints " +
+                "where table_name='t'" +
+                "order by constraint_name";
+        execute(selectCheckConstraintsStmt);
+        assertThat(printedTable(response.rows()), is(
+            "doc| t| CHECK| check_id_gt_zero\n" +
+                "doc| t| CHECK| check_x_gt_zero\n" +
+                "doc| t| CHECK| doc_t_id_not_null\n" +
+                "doc| t| PRIMARY KEY| t_pk\n"
+        ));
+
+        execute("alter table t drop constraint check_x_gt_zero");
+        refresh();
+
+        execute(selectCheckConstraintsStmt);
```

Comment:
could use a `WHERE constraint_name = 'check_x_gt_zero` filter to be more concrete.